File: AgendaRails/app/models/contato.rb
class Contato < ApplicationRecord
  validates :nome, presence: true, length: {minimum: 5}
  validates :endereco, presence: true
  validates :telefone, length: {minumum: 10, maximum: 11}
  validates :email, format: { with: /\A([^@\s]+)@((?:[-a-z0-9]+\.)+[a-z]{2,})\z/i, on: :create }
end

File: agenda-server/app_api/models/contato.js
var mongoose = require('mongoose');

var contato = new mongoose.Schema({
    nome: {
        type: String,
        required: true
    },
    telefone: {
        type: String,
        required: true
    },
    endereco: {
        type: String
    }
});

module.exports = mongoose.model('Contato', contato);